### PR TITLE
LL-1870 - React Native 0.61

### PR DESCRIPTION
**Pull request by MortalKastor**
## Upgrade to the latest React Native.

This update is quite invasive as the native dependencies linking system changed, bringing quite some changes to our build scripts.

That's for the best though, as it brings [autolinking](https://facebook.github.io/react-native/blog/2019/07/03/version-60#native-modules-are-now-autolinked) 🎉

### Opinionated changes

I arbitrarily chose to add the default React Native eslint config to our ruleset, with some rules which were triggered by our code turned off (because they're about conflicting code style) and some other moved to warn (because they're arguably good practice, but can wait). See `.eslintrc.js`

### To do

- [x] Update libcore for iOS LedgerHQ/lib-ledger-core-react-native-bindings#36
- [x] Get it to compile for iOS
- [x] Get it to run on iOS
- [x] Update libcore for Android LedgerHQ/lib-ledger-core-react-native-bindings#37 + LedgerHQ/djinni#36
- [x] Get it to compile for Android
- [x] Get it to run on Android
- [x] Update [Lean Core](https://github.com/facebook/react-native/issues/23313) dependencies (maybe use [rn-update-deprecated-modules](https://github.com/lucasbento/rn-update-deprecated-modules))
- [ ] Fix flowtype errors and eslint warnings
- [ ] Make sure no screen is broken 💔

### Type

Tech stack update

### Context

https://ledgerhq.atlassian.net/browse/LL-1870
https://ledgerhq.atlassian.net/browse/LLC-340
LedgerHQ/lib-ledger-core-react-native-bindings#36
LedgerHQ/lib-ledger-core-react-native-bindings#37

### Parts of the app affected / Test plan

Basically all the app is affected by this since there are some big changes around how native libs are linked, sorry 🙇

### CODE REVIEW

**Comment by gre** [comment]
👏  🚀

**Comment by MortalKastor** [comment]
I played a bit with app on both Android and iOS, installing apps, importing accounts, sending and receiving funds etc, on BLE and on USB and I finally got rid of all the (many) crashes I could stumble upon.

There still are some Flow errors, and probably some glitches (like the recipient address field on Android being 1 line high, hence showing only part of the address)
